tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+    - component: Device:Startup
+```
+
+## Software Layers
+
+Software layers collect source files and software components along with configuration files for re-use in different projects.
+The following diagram shows the various layers that are used to compose the IoT Cloud examples.
+
+![Software Layers](./images/Layer.png "Target and Build Types")
+
+The following example is a `Blinky` application that uses a `App`, `Board`, and `RTOS` layer to compose the application for a NUCELO-G474RE board.  Note, that the `device:` definition is is the `Board` layer.
+
+**Example Project: `Blinky.cproject.yml`**
+
+```yml
+project:
+  compiler: AC6
+
+  layers:
+    - layer: .\Layer\App\Blinky.clayer.yml
+    - layer: .\Layer\RTOS\RTX.clayer.yml
+    - layer: .\Layer\Board\Nucleo-G474RE.clayer.yml
+```
+
+**App Layer: `.\Layer\App\Blinky.clayer.yml`**
+
+```yml
+layer:
+# type: RTOS
+  name: RTX
+  description: Keil RTX5 open-source real-time operating system with CMSIS-RTOS v2 API
+
+  interfaces:
+    - provides:
+        - RTOS2:
+
+  components:
+    - component: CMSIS:RTOS2:Keil RTX5&Source
+```
+
+**RTOS Layer: `.\Layer\RTOS\RTX.clayer.yml`**
+
+```yml
+layer:
+# type: RTOS
+  name: RTX
+  description: Keil RTX5 open-source real-time operating system with CMSIS-RTOS v2 API
+
+  interfaces:
+    - provides:
+        - RTOS2:
+
+  components:
+    - component: CMSIS:RTOS2:Keil RTX5&Source
+```
+
+**Board Layer: `.\Layer+NUCELO-G474RE\Board\Nucleo-G474RE.clayer.yml`**
+
+```yml
+layer:
+  name: NUCLEO-G474RE
+# type: Board
+  description: Board setup with interfaces
+  device: STM32G474CBTx
+
+  interfaces:
+    - consumes:
+        - RTOS2:
+    - provides:
+        - C_VIO:
+        - A_IO9_I:
+        - A_IO10_O:
+        - C_VIO:
+        - STDOUT:
+        - STDIN:
+        - STDERR:
+        - Heap: 65536
+  
+  components:
+    - component: CMSIS:CORE
+    - component: CMSIS Driver:USART:Custom
+    - component: CMSIS Driver:VIO:Board&NUCLEO-G474RE
+    - component: Compiler&ARM Compiler:Event Recorder&DAP
+    - component: Compiler&ARM Compiler:I/O:STDERR&User
+    - component: Compiler&ARM Compiler:I/O:STDIN&User
+    - component: Compiler&ARM Compiler:I/O:STDOUT&User
+    - component: Board Support&NUCLEO-G474RE:Drivers:Basic I/O
+    - component: Device&STM32CubeMX:STM32Cube Framework:STM32CubeMX
+    - component: Device&STM32CubeMX:STM32Cube HAL
+    - component: Device&STM32CubeMX:Startup
+  
+  groups:
+    - group: Board IO
+      files:
+        - file: ./Board_IO/arduino.c
+        - file: ./Board_IO/arduino.h
+        - file: ./Board_IO/retarget_stdio.c
+    - group: STM32CubeMX
+      files:
+        - file: ./RTE/Device/STM32G474RETx/STCubeGenerated/STCubeGenerated.ioc
+```
+
+## Project Setup for Multiple Targets and Builds
+
+Complex examples require frequently slightly different targets and/or modifications during build, i.e. for testing. The picture below shows a setup during software development that supports:
```

Comment:
Here is a subtlety that may not be one later: "slightly different targets"  ... is this something that matter ? If yes, how do we ensure that ? If not, then do we need to mention that